src/test/java/uk/gov/pay/adminusers/service/SecondFactorAuthenticatorTest.java

```diff
@@ -28,6 +30,24 @@ public void shouldGenerateAndValidate2FAPasscode() throws Exception {
         assertFalse(secondFactorAuthenticator.authorize(secret + 1, password));
     }
 
+    @Test
+    public void shouldSuccess_ifAskedToValidateLastSteps2FAPasscode() throws Exception {
+        String secret = "mysecret";
+        Integer password = secondFactorAuthenticator.newPassCode(secret);
+        TimeUnit.SECONDS.sleep(30);
```

Comment:
Would it be possible to inject, mock, stub the time for this to avoid sleeps @maxcbc, even a tacky optional override into the authorisze method with the time to use rather than using `System.currentTimeMillis()`